core/utils/pcap_handle.h

```diff
@@ -0,0 +1,39 @@
+#ifndef PCAP_HANDLE_H
+#define PCAP_HANDLE_H
+
+#include <errno.h>
+#include <pcap/pcap.h>
+
+#include <glog/logging.h>
+
+#include "../message.h"
+#include "../port.h"
+#include "../snobj.h"
+#include "../utils/pcap.h"
+
+// Wraps all accesses to libpcap; opens a device, sets it nonblocking, and then
+// lets you send/recv packets.
```

Comment:
I would move the comment text after the semicolon to the constructor / other methods.